server/server.go

```diff
@@ -821,7 +832,9 @@ func (s *StanServer) createNatsClientConn(name string, sOpts *Options, nOpts *se
 		ncOpts.ReconnectBufSize = 128
 	}
 
-	Tracef(" NATS conn opts: %v", ncOpts)
+	if s.trace {
```

Comment:
For performance. Early on, I realized that the cost in calling Tracef or Debugf is substantial in term of string allocation (if there is formatting involved, etc..), so it was better to do the check at the higher layer. I have kept the check at the lower level in case one forgets to do the check before calling those functions, and to respect the contract of the logger.